public/posts/v1.7.md

```diff
@@ -0,0 +1,465 @@
+
+Here is an example where we create a executable from the std/http/file_server
+module that listens on port 8080 (instead of default 4507), and has CORS
+enabled. The permissions that the running code has is also locked down (can only
+read from current working directory, and can only listen on port 8080).
+
+<img src="/posts/v1.7/compile_with_args.gif">
+
+## Support for importing data URLs
+
+[Data URLs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs)
+are a useful tool for executing code generated on the fly. In this release we
+added support for data URLs in imports (both static and dynamic), as well as in
+Web Workers. This feature is already supported across all modern browsers and
+NodeJS.
+
+Here is an example for you to try out:
+
+```ts
+// main.ts
+export const a = "a";
+
+export enum A {
+  A,
+  B,
+  C,
+}
+```
+
+Above code can be expressed as following data URL:
+`"data:application/typescript;base64,ZXhwb3J0IGNvbnN0IGEgPSAiYSI7CgpleHBvcnQgZW51bSBBIHsKICBBLAogIEIsCiAgQywKfQo="`.
+This URL is created by base64 encoding the file contents, and appending it to
+`data:application/typescript;base64,`. For JavaScript you would append the
+contents to `data:application/javascript;base64,`.
+
+This import specifier can later be imported like so:
+
+```ts
+// https://deno.land/posts/v1.7/import_data_url.ts
+import * as a from "data:application/typescript;base64,ZXhwb3J0IGNvbnN0IGEgPSAiYSI7CgpleHBvcnQgZW51bSBBIHsKICBBLAogIEIsCiAgQywKfQo=";
+
+console.log(a.a);
+console.log(a.A);
+console.log(a.A.A);
+```
+
+Try it:
+
+```
+$ deno run https://deno.land/posts/v1.7/import_data_url.ts
+a
+{ "0": "A", "1": "B", "2": "C", A: 0, B: 1, C: 2 }
+0
+```
+
+Similarly for workers:
+
+```ts
+// https://deno.land/posts/v1.7/worker_data_url.ts
+import { deferred } from "https://deno.land/std@0.83.0/async/deferred.ts";
+import { assertEquals } from "https://deno.land/std@0.83.0/testing/asserts.ts";
+
+const promise = deferred();
+const tsWorker = new Worker(
+  `data:application/typescript;base64,${btoa(`
+    if (self.name !== "tsWorker") {
+      throw Error(\`Invalid worker name: \${self.name}, expected tsWorker\`);
+    }
+    onmessage = function (e): void {
+      postMessage(e.data);
+      close();
+    };
+  `)}`,
+  { type: "module", name: "tsWorker" }
+);
+
+tsWorker.onmessage = (e): void => {
+  assertEquals(e.data, "Hello World");
+  promise.resolve();
+};
+
+tsWorker.postMessage("Hello World");
+
+await promise;
+tsWorker.terminate();
+```
+
+## New unstable `Deno.resolveDns` API
+
+This release adds a new `Deno.resolveDns` API. It can be used to query DNS
+records from a DNS resolver. At the moment only DNS over UDP is supported (no
+DNS over HTTPS nor DNS over TLS). One can specify a custom nameserver (for
+example Cloudflare's 1.1.1.1 or Google's 8.8.8.8) to use, but by default we will
+use the system resolver (e.g. `/etc/resolv.conf` on Linux).
```

Comment:
```suggestion
This release adds a new `Deno.resolveDns` API. It can be used to query DNS
records from a DNS resolver. At the moment only DNS over UDP/TCP is supported (no
DNS over HTTPS nor DNS over TLS). One can specify a custom nameserver (for
example Cloudflare's 1.1.1.1 or Google's 8.8.8.8) to use, but by default we will
use the system resolver (e.g. `/etc/resolv.conf` on Linux).
```

Actually, lookup through TCP is also supported. (In fact, the integration test would pass with [this line](https://github.com/denoland/deno/blob/master/cli/tests/integration_tests.rs#L5553) commented out).
At first, I thought it would be good to have option for the users to specify which protocol to use. But it became impossible because I decided to use [this method of trust_dns_resolver](https://docs.rs/trust-dns-resolver/0.20.0/trust_dns_resolver/config/struct.NameServerConfigGroup.html#method.from_ips_clear) and it automatically creates both TCP and UDP connections internally.